File: pyStudy/day0102/study04.py
# 사칙 연산 클래스
#               + , -  , * , /            입력받는 메서드
# 메서드가 4개(add, sub, mul, div) + 메서드 1개(setData)

class FourCal:

    def __init__(self, first, second):      # 생성자  -> 파이썬에서 이렇게 사용함
        self.first = first;
        self.second = second;

    def setData(self, first, second):
        self.first = first;
        self.second = second;

    def add(self):
        result = self.first + self.second;
        return result;

    def sub(self):
        result = self.first - self.second;
        return result;

    def mul(self):
        result = self.first * self.second;
        return result;

    def div(self):
        result = self.first / self.second;
        return result;


# a = FourCal();
# a.setData(4, 2);

a = FourCal(4,2);

print(a.add());     # 결과 6;   
print(a.sub());     # 결과 2;
print(a.mul());     # 결과 8;
print(a.div());     # 결과 2;

File: pyStudy/day0102/Study04.java
package day0102;

public class Study04 {
    public static void main(String[] args) {
        
        FourCal a = new FourCal();
        // a.setData(4,2);   생성자를 추가 안했을 경우 -> 메서드를 만들어서 사용한 것
        
        System.out.println(a.add());
        System.out.println(a.sub());
        System.out.println(a.mul());
        System.out.println(a.div());

    }
}
